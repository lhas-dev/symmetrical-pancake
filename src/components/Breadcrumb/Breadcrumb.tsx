import { Icon } from "../Icon/Icon";
import styled from "styled-components";
import { useDevice } from "../../hooks/useDevice";

interface ItemProps {
  label: string;
}

interface BreadcrumbProps {
  items: ItemProps[];
}

const Wrapper = styled.ul`
  margin: 0;
  padding: 0;
  display: flex;
  align-items: center;
  list-style-type: none;
`;

const Item = styled.li`
  display: flex;

  .ArrowRightIcon {
    margin-left: 14px;
    margin-right: 14px;
  }
`;

const Link = styled.a`
  margin: 0;
  font-family: "Open Sans", sans-serif;
  font-weight: 600;
  font-size: 11px;
  line-height: 15px;
  text-transform: uppercase;
  letter-spacing: 1.5px;
  text-decoration: none;
  color: #272727;
`;

export const Breadcrumb = ({ items }: BreadcrumbProps) => {
  const device = useDevice();

  return (
    <Wrapper>
      <Item>
        <Link href="#">
          {device.isDesktop() && "In√≠cio"}
          {device.isMobile() && <Icon src="EtceteraIcon" />}
        </Link>
        <Icon src="ArrowRightIcon" />
      </Item>
      {items.map(({ label }) => (
        <Item key={label}>
          <Link href="#">{label}</Link>
        </Item>
      ))}
    </Wrapper>
  );
};
