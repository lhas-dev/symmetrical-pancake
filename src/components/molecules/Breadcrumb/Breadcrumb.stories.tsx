import { ComponentStory, ComponentMeta } from "@storybook/react";

import { Breadcrumb } from "./Breadcrumb";

export default {
  title: "Storybook/Molecules/Breadcrumb",
  component: Breadcrumb,
  argTypes: {
    src: { control: "text" },
  },
} as ComponentMeta<typeof Breadcrumb>;

const Template: ComponentStory<typeof Breadcrumb> = (args) => (
  <Breadcrumb {...args} />
);

export const Default = Template.bind({});
Default.args = {
  items: [
    {
      label: "PÃ¡gina",
    },
  ],
};
