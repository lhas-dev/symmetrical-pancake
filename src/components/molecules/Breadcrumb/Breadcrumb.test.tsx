import { render, act } from "@testing-library/react";
import { Breadcrumb } from "./Breadcrumb";

test("renders Breadcrumb with valid params on mobile devices", async () => {
  Object.defineProperty(window.navigator, "userAgent", {
    value: "Android",
    writable: true,
  });

  const items = [
    {
      label: "Página",
    },
  ];

  const tree = await act(async () => {
    render(<Breadcrumb items={items} />);
  });
  expect(tree).toMatchSnapshot();
});

test("renders Breadcrumb with valid params on desktop devices", async () => {
  Object.defineProperty(window.navigator, "userAgent", {
    value: "Chrome",
    writable: true,
  });

  const items = [
    {
      label: "Página",
    },
  ];

  const tree = await act(async () => {
    render(<Breadcrumb items={items} />);
  });
  expect(tree).toMatchSnapshot();
});
