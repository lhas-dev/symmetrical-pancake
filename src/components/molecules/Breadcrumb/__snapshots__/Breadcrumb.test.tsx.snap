// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Breadcrumb with valid params on desktop devices 1`] = `undefined`;

exports[`renders Breadcrumb with valid params on mobile devices 1`] = `undefined`;
